pkg/operator/ceph/csi/peermap/clientProfile.go

```diff
@@ -0,0 +1,162 @@
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package peermap
+
+import (
+	"context"
+	"reflect"
+	"slices"
+
+	csiopv1a1 "github.com/ceph/ceph-csi-operator/api/v1alpha1"
+	"github.com/pkg/errors"
+	cephv1 "github.com/rook/rook/pkg/apis/ceph.rook.io/v1"
+	"github.com/rook/rook/pkg/clusterd"
+	cephclient "github.com/rook/rook/pkg/daemon/ceph/client"
+	kerrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// ReconcileClientProfileMappings creates/updates a clientProfileMappings resource used by the ceph csi-operator
+func ReconcileClientProfileMappings(ctx context.Context, clusterContext *clusterd.Context, clusterInfo *cephclient.ClusterInfo, pool *cephv1.CephBlockPool) error {
+	if pool.Spec.Mirroring.Peers == nil {
+		logger.Debugf("no peer secrets added in cephBlockPool %q. skip creating clientProfileMappings resource", pool.Name)
+		return nil
+	}
+
+	newClientProfileMappings, err := generateClientProfileMappingsCR(clusterContext, clusterInfo, pool)
+	if err != nil {
+		return errors.Wrapf(err, "failed to generate clientProfileMappings for cephBlockPool %q", pool.Name)
+	}
+
+	err = createOrUpdateClientProfileMappingsCR(clusterContext, clusterInfo, newClientProfileMappings)
+	if err != nil {
+		return errors.Wrapf(err, "failed to create or update clientProfileMappings resource for cephBlockPool %q", pool.Name)
+	}
+
+	return nil
+}
+
+func createOrUpdateClientProfileMappingsCR(clusterContext *clusterd.Context, clusterInfo *cephclient.ClusterInfo, clientProfileMappings *csiopv1a1.ClientProfileMapping) error {
+	existingClientProfileMappings := &csiopv1a1.ClientProfileMapping{}
+	request := types.NamespacedName{Name: clientProfileMappings.Name, Namespace: clusterInfo.Namespace}
+	err := clusterContext.Client.Get(clusterInfo.Context, request, existingClientProfileMappings)
+	if err != nil {
+		if kerrors.IsNotFound(err) {
+			err := clusterContext.Client.Create(clusterInfo.Context, clientProfileMappings)
+			if err != nil {
+				return errors.Wrapf(err, "failed to create clientProfileMappings CR %q", clientProfileMappings.Name)
+			}
+			logger.Infof("successfully created clientProfileMappings CR %q", clientProfileMappings.Name)
+			return nil
+		}
+		return errors.Wrapf(err, "failed to get existing clientProfileMappings CR %q", clientProfileMappings.Name)
+	}
+
+	if reflect.DeepEqual(existingClientProfileMappings.Spec, clientProfileMappings.Spec) {
+		logger.Debug("no change in client profile mappings. Skipping update of clientProfileMappings resource")
+		return nil
+	}
+
+	existingClientProfileMappings.Spec = clientProfileMappings.Spec
+	err = clusterContext.Client.Update(clusterInfo.Context, existingClientProfileMappings)
+	if err != nil {
+		return errors.Wrapf(err, "failed to update existing clientProfileMappings CR  %q", clientProfileMappings.Name)
```

Comment:
```suggestion
return errors.Wrapf(err, "failed to update existing clientProfileMappings %q", clientProfileMappings.Name)
```

can we remove CR from all the logs and error messages as well?